drivers/flash/nrf_qspi_nor.c

```diff
@@ -311,23 +311,32 @@ static int qspi_send_cmd(const struct device *dev, const struct qspi_cmd *cmd,
 		return -EINVAL;
 	}
 
-	qspi_lock(dev);
+	size_t xfer_len = sizeof(cmd->op_code);
+
+	if (cmd->tx_buf != NULL) {
+		xfer_len += cmd->tx_buf->len;
+	}
+
+	if (cmd->rx_buf != NULL) {
+		xfer_len += cmd->rx_buf->len;
+	}
+
+	if (xfer_len > NRF_QSPI_CINSTR_LEN_9B) {
+		LOG_WRN("cinstr %02x transfer too long: %zu",
+			cmd->op_code, xfer_len);
+		return -EINVAL;
+	}
```

Comment:
So the command send + the data to command + response data can not be longer than NRF_QSPI_CINSTR_LEN_9B?
Seems that within nrfx_qspi_cinstr_xfer the nrf_qspi_cinstrdata_set and nrf_qspi_cinstrdata_get both use xfer_len (p_config->length).

Reply:
Yes, because like most SPI interfaces the transfer is bidirectional.  If you actually provide buffers for both send and receive, they have to be the same length: the output signal comes from the tx and whatever's received during the transfer is stored into rx.  The transfer is not ntx bytes send followed by nrx bytes of read.

Reply:
Ah right.